galaxychop/galaxychop.py

```diff
@@ -92,54 +92,127 @@ class Galaxy:
+    of stars, gas and DM.
+   
+    Parameters
+    ----------
+   
+    pos_star_rot, vel_star_rot. 
+    pos_dark_rot, vel_dark_rot. 
+    pos_gas_rot, vel_gas_rot.
+   
+    Returns
+    -------
+   
+   L_part = np.array(3, 3*n), concatenates angular momentum for the stars, 
+   dark matter and gas in a single array.
+   
+   Lr_star = np.array(n, 1), Component in the plane of the angular 
+   momentum of stars. 
+   
+   Lr = np.array(3*n, 1) Component in the plane of the angular 
+   momentum for all particles.
+   
+    """
+
+    L_dark = np.asarray((pos_dark_rot[:, 1] * vel_dark_rot[:, 2] -
+                         pos_dark_rot[:, 2] * vel_dark_rot[:, 1],
+                         pos_dark_rot[:, 2] * vel_dark_rot[:, 0] -
+                         pos_dark_rot[:, 0] * vel_dark_rot[:, 2],
+                         pos_dark_rot[:, 0] * vel_dark_rot[:, 1] -
+                         pos_dark_rot[:, 1] * vel_dark_rot[:, 0]))
+
+    L_star = np.asarray((pos_star_rot[:, 1] * vel_star_rot[:, 2] -
+                         pos_star_rot[:, 2] * vel_star_rot[:, 1],
+                         pos_star_rot[:, 2] * vel_star_rot[:, 0] -
+                         pos_star_rot[:, 0] * vel_star_rot[:, 2],
+                         pos_star_rot[:, 0] * vel_star_rot[:, 1] -
+                         pos_star_rot[:, 1] * vel_star_rot[:, 0]))
+
+    L_gas = np.asarray((pos_gas_rot[:, 1] * vel_gas_rot[:, 2] - 
+                        pos_gas_rot[:, 2] * vel_gas_rot[:, 1],
+                        pos_gas_rot[:, 2] * vel_gas_rot[:, 0] - 
+                        pos_gas_rot[:, 0] * vel_gas_rot[:, 2],
+                        pos_gas_rot[:, 0] * vel_gas_rot[:, 1] - 
+                        pos_gas_rot[:, 1] * vel_gas_rot[:, 0]))
 
     L_part = np.concatenate((L_gas, L_dark, L_star), axis=1)
 
-    # Componente en el plano del momento angular de estrellas.
-    Lr_star = np.sqrt(L_star[0, :]**2 + L_star[1, :]**2)
-    # Componente en el plano del momento angular de las particulas.
-    Lr = np.sqrt(L_part[0, :]**2 + L_part[1, :]**2)
+    Lr_star = np.sqrt(L_star[0, :] ** 2 + L_star[1, :] ** 2)
+
+    Lr = np.sqrt(L_part[0, :] ** 2 + L_part[1, :] ** 2)
+
+    """Calculation of kinetic energy
 
-    # Calculamos la energia cinetica de las partículas.
-    k_star = 0.5*(star[:, 4]**2 + star[:, 5]**2 + star[:, 6]**2)
-    k_dark = 0.5*(dark[:, 4]**2 + dark[:, 5]**2 + dark[:, 6]**2)
-    k_gas = 0.5*(gas_[:, 4]**2 + gas_[:, 5]**2 + gas_[:, 6]**2)
+    Parameters
+    ----------
+   
+    vx_s, vy_s, vz_s.
+    vx_dm, vy_dm, vz_dm.
+    vx_g, vy_g, vz_g.
+   
+    Returns
+    -------
+   
+    k_star = np.array(n,1), kinetic energy for the stars 
+    k_dark = np.array(m,1), kinetic energy for the dark matter
+    k_gas = np.array(g,1), kinetic energy for the gas
+   
+    """
 
+    k_star = 0.5 * (vx_s ** 2 + vy_s ** 2 + vz_s ** 2)
+    k_dark = 0.5 * (vx_dm ** 2 + vy_dm ** 2 + vz_dm ** 2)
+    k_gas = 0.5 * (vx_g ** 2 + vy_g ** 2 + vz_g ** 2)
+   
```

Comment:
This corresponds to issue #49 because you are calculating kinetic energy.

Reply:
I might see a conflict here with Antonellas PR : #58